lib/backend/etcdv3/etcdv3.go

```diff
@@ -404,6 +404,21 @@ func (c *etcdV3Client) Clean() error {
 	return nil
 }
 
+// IsClean() returns true if there are no /calico/ prefixed entries in the
+// datastore.  This is not part of the exposed API, but is public to allow
+// direct consumers of the backend API to access this.
+func (c *etcdV3Client) IsClean() (bool, error) {
+	log.Debug("Calling Get on etcdv3 client")
+	resp, err := c.etcdClient.Get(context.Background(), "/calico/", clientv3.WithPrefix())
+	if err != nil {
+		log.WithError(err).Info("Error returned from etcdv3 client")
```

Comment:
Shouldn't this be an error log level or a warning at least?

Reply:
no - that's up to the end user if they want to log it as an error

generally lower down the stack you keep your levels more as debug or info.